test.py

```diff
@@ -1349,6 +1352,8 @@ def parse_cmd_line() -> argparse.Namespace:
     parser.add_argument("--artifacts_dir_url", action='store', type=str, default=None, dest="artifacts_dir_url",
                         help="Provide the URL to artifacts directory to generate the link to failed tests directory "
                              "with logs")
+    parser.add_argument("--ci", action="store_true", default=False, dest="ci",
+                        help="Say explicitly if the execution is on CI to use another pool size quantity")
     scylla_additional_options = parser.add_argument_group('Additional options for Scylla tests')
```

Comment:
I think we can determine the pool size from the machine. Assume each cluster takes 2 vcpu (usually not all vcpus will be active in our tests); and the amount of memory is known from the startup parameters.

Reply:
Potentially yes, but we do not know who else is running right now, especially that not all tests are using this method to start clusters. We know this on test.py level, but we cannot change the pool size once it created and cannot change it when another thread will finish.

Reply:
@avikivity. The problem is that pool sizes work between tests within the folder. but not between folders.  Your suggestion will work if we have one common pool, but now it is just in the roadmap because requires a lot of changes in the framework.

now we just try to speed ci using what test.py already has with minimal changes/efforts.
and looks like it works. 

without --ci - test.py runs took  2h 1min 47s - https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/10501/execution/node/681/log/

with --ci  - test.py runs took 1h 10min 58s - https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/10509/execution/node/683/log/?consoleFull

so almost 2 times faster just for +10 -1 change 

cc @kostja, @nyh

Reply:
Do we issue --ci today?

Reply:
I guess we don't. Why don't we change the configuration globally (not just with --ci)?

Reply:
I approved this PR but please resolve with Avi this question first.

Reply:
We do not use it. My laptop will hang with 10 clusters for topology. So I don't want to break local execution. This is just a very first step to scheduling based on resources automatically. It reduces a little bit the pain of waiting for the test result while I can continue to work on a better solution.

Reply:
I would like to avoid introducing the new --ci parameter - it's yet another thing that will be different from CI and the regular developer workflow.

Most developers have a workstation, maybe we can invert the sense and introduce --cluster-pool-size, with the default suitable for CI and workstations, and non-default for laptops.

Reply:
We can leave the current number of clusters as it is and introduce the parameter for test.py --cluster-pool-size as you suggest. So we can set it during the start of the test.py on CI and locally someone will be able to increase the pool size without touching the code.

Reply:
but for different Test suits poll_size can have different optimal number

Reply:
@avikivity we already have --ci which is --verbose; and a bunch of options for junit output, which are not enabled in --ci; we have been avoiding --ci on numerous occasions. Here, a single switch changes sets its own pool size in every suite. We could have some template file where we would derive the pool size from, and collect all pool sizes in this file. This would add a yet another config file.

Reply:
@avikivity correct if I'm wrong, you want to use the pool size configured for CI and use --cluster-pool-size to set the pool size from the command line when we are executing on less powerful machine?

Reply:
@avikivity I don't know if really "most developers have workstations". I run all tests on my laptop, and I bet that if you take a poll you'll see that many if not most developers do that.
I personally think (and try to repeat this every time I get a chance, so sorry ;-)) that testing should be **convenient, easy and fast for developers**. I personally think we should force the CI to pass ugly parameters that nobody remembers - NOT developers - developers should not need to remember bizarre parameters to get tests running.

Reply:
> After an email conversation with @avikivity, I've inverted the logic: so now pool sizes are configured by default for CI/workstations, and additional parameter or env var to reduce pool sizes on weaker machines.

I find myself compelled that despite this being Avi's long-standing position, I think he has been wrong here and still wrong. Avi, I think this approach just makes it harder for developers to work naturally. You are hoping that developers will be forced to use the inappropriate default which CI wants, but even more likely is that the developers will just need to memorize (or create a script) bizarre options to override your decisions. As an example, years after NOBODY uses DPDK with Scylla, building with DPDK is still our default and I still need to run "./configure.py --disable-dpdk" every time to avoid wasting time with DPDK.

Reply:
I don't understand why --ci and --verbose are related.

Reply:
> @avikivity I don't know if really "most developers have workstations". I run all tests on my laptop, and I bet that if you take a poll you'll see that many if not most developers do that. I personally think (and try to repeat this every time I get a chance, so sorry ;-)) that testing should be **convenient, easy and fast for developers**. I personally think we should force the CI to pass ugly parameters that nobody remembers - NOT developers - developers should not need to remember bizarre parameters to get tests running.

Why are you subjecting yourself to such pain?